crates/modelardb_server/src/parser.rs

```diff
@@ -874,6 +875,68 @@ mod tests {
         assert_eq!(error.unwrap_err(), expected_error);
     }
 
+    #[test]
+    fn test_tokenize_and_parse_create_table_with_lowercase_keyword_as_table_name() {
```

Comment:
Since I have to give format a string literal with the `{}` at compile-time, I have to do two separate functions anyway: one for model table and one for normal table. Or am I missing something? As far as I am aware, I can't use variables as the first argument in the formatter.

Reply:
It is correct that `format!()` must be given a string literal as its first argument, however, in this case the formatting is just a simple substitution so as far as I can tell we can simply replace it with [`str::replace()`](https://doc.rust-lang.org/std/primitive.str.html#method.replace) like [for example:](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021) `let sql_with_table = sql_without_table.replace("{}", "table_name");`.

Reply:
Ah, I see. Something like this?

Reply:
Yes, but I suggest also moving the loops to `parse_and_assert_if_keyword_is_restricted_in_table_name()` so it contains all of the duplicate code. This should be simple to do if the parameter `keyword` in `parse_and_assert_if_keyword_is_restricted_in_table_name()` is replaced with a parameter that accept a function from `&str` to `String` so something like `|str| str.to_lowercase()` and `|str| str.to_uppercase()` can be passed.

Reply:
I am not hugely familiar with higher order functions in Rust, but is this the correct implementation?

Reply:
Yes, but it is not necessary to create an anonymous function to pass the functions as they match the type you specified for the first argument of `parse_and_assert_if_keyword_is_restricted_in_table_name`. Instead you can pass it as a direct reference to the method like this `str::to_lowercase`.

Reply:
Fixed.